### PR TITLE
Add logging to keen analysis scripts [PLAT-1023]

### PR DESCRIPTION
**Pull request by erinspace**
## Purpose

Figuring out what was actually done when running Keen analytics after the fact is hard as there was no file logger – add one and use it to log all keen analytic script logs to a file!

## Changes

- change file logger in `scripts/analysis/base.py` to be for all analysis scripts
- add file logger that will log all child loggers in `scripts/analysis`
- small script improvements discovered with local testing

## QA Notes

No QA needed!

## Documentation
None necessary

## Side Effects
None anticipated

## Ticket
https://openscience.atlassian.net/browse/PLAT-1023

### CODE REVIEW

File: scripts/analytics/base.py [hunk 1]
```diff
@@ -9,10 +9,13 @@
 from website.app import init_app
 from website.settings import KEEN as keen_settings
 from keen.client import KeenClient
+from scripts import utils as script_utils
 
-logger = logging.getLogger(__name__)
+logger = logging.getLogger('scripts.analytics')
 logging.basicConfig(level=logging.INFO)
 
+script_utils.add_file_logger(logger, 'keen_analytics')
```
Comment (sloria) [inline scripts/analytics/base.py]: Nit: We should generally avoid import side-effects, i.e. importing a module shouldn't create file. So the file logger should be created in the `run_main` function for the scripts being run.

File: scripts/analytics/user_summary.py [hunk 2]
```diff
@@ -117,7 +118,7 @@ def get_events(self, date):
         try:
             # Because this data reads from Keen it could fail if Keen read api fails while writing is still allowed
             counts['status']['stickiness'] = self.calculate_stickiness(timestamp_datetime, query_datetime)
-        except requests.exceptions.ConnectionError:
+        except (requests.exceptions.ConnectionError, keen_exceptions.InvalidProjectIdError):
```
Comment (sloria) [inline scripts/analytics/user_summary.py]: 👍

File: scripts/analytics/run_keen_summaries.py [hunk 3]
```diff
@@ -8,6 +9,10 @@
 from scripts.analytics.preprint_summary import PreprintSummary
 from scripts.analytics.institution_summary import InstitutionSummary
 from scripts.analytics.base import DateAnalyticsHarness
+from scripts.utils import add_file_logger
+
+logger = logging.getLogger('scripts.analytics')
+logging.basicConfig(level=logging.INFO)
```
Comment (sloria) [inline scripts/analytics/run_keen_summaries.py]: Why was this added?

Comment (erinspace) [reply scripts/analytics/run_keen_summaries.py]: I suppose the log level isn't needed, actually, but I added the file logger in these to address the comment "the file logger should be created in the run_main function for the scripts being run," and I needed a logger to pass into the individual summary scripts that were being run by celery

Comment (sloria) [reply scripts/analytics/run_keen_summaries.py]: I was only referring to the `basicConfig` call. Can it be removed?

Comment (erinspace) [reply scripts/analytics/run_keen_summaries.py]: Ah ok! Yes and removed!
